Description: [droidcamsrc] don't fail device switch while running
 This matches the behavior of gstahc, a camerabin-compatible element for
 Android's Camera API, which is in gst-plugins-bad upstream. The set
 device will take effect next time the camera is opened.
 .
 A useless call to gst_droidcamsrc_photography_init() is also removed.
 That function does nothing if src->photo is already initialized.
Author: Ratchanan Srirattanamet <ratchanan@ubports.com>
Forwarded: no
Last-Update: 2020-06-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/gst/droidcamsrc/gstdroidcamsrc.c
+++ b/gst/droidcamsrc/gstdroidcamsrc.c
@@ -341,15 +341,14 @@
 
   switch (prop_id) {
     case PROP_CAMERA_DEVICE:
+      src->camera_device = g_value_get_enum (value);
       g_rec_mutex_lock (&src->dev_lock);
       if (src->dev && src->dev->info) {
-        GST_ERROR_OBJECT (src,
-            "cannot change camera-device while camera is running");
+        GST_WARNING_OBJECT (src,
+            "changing camera-device to %d will take effect next time the camera is opened.",
+            src->camera_device);
       } else {
-        src->camera_device = g_value_get_enum (value);
         GST_INFO_OBJECT (src, "camera device set to %d", src->camera_device);
-        /* initialize empty photo properties */
-        gst_droidcamsrc_photography_init (src);
       }
       g_rec_mutex_unlock (&src->dev_lock);
       break;
